bug: server crash on startup — Eio_mutex.Poisoned after cascade refactor

## 증상
서버 시작 즉시 crash:
```
[ERROR] [Server] [main] keeper bootstrap failed: Cancelled: Eio__Eio_mutex.Poisoned(_)
masc-mcp: internal error, uncaught exception: Eio__Eio_mutex.Poisoned(_)
```

## 재현
```bash
git checkout main  # d97dca92
dune build --root .
./start-masc-mcp.sh --http --port 8935  # crash
```

## 정상 동작 커밋
```bash
git checkout 02da2c68  # v2.129.0
# 이 커밋에서는 서버 정상 시작
```

## 원인 추정
v2.129.0 (02da2c68) 이후 머지된 커밋 중 cascade refactor:
- #2325 (Phase 2: swap parsing boundary)
- #2329 (Phase 3: delegate default/preset resolution)

이 커밋들이 keeper bootstrap 경로에서 OAS Cascade_config를 호출하는데, Eio context 밖에서 Eio.Mutex를 사용하거나 초기화 순서 문제로 Poisoned 발생 추정.

## 현재 상태
서버를 v2.129.0 바이너리로 롤백하여 운영 중.

--- a/lib/model_spec.ml
+++ b/lib/model_spec.ml
@@ -344,7 +344,9 @@ let resolve_cascade_labels ~name ~defaults =
         Llm_provider.Cascade_config.resolve_model_strings
           ~config_path ~name ~defaults ()
       with
+      | Eio.Cancel.Cancelled (Eio.Mutex.Poisoned _) -> defaults
       | Eio.Cancel.Cancelled _ as e -> raise e
+      | Effect.Unhandled _ | Eio.Mutex.Poisoned _ -> defaults
       | _ ->
           (* Eio.Mutex requires Eio context; gracefully fall back when
              called during module init (before Eio.main). *)

